op-e2e/actions/proofs/jovian_activation_test.go

```diff
@@ -0,0 +1,210 @@
+
+	runJovianDerivationTest := func(gt *testing.T, testCfg *helpers.TestCfg[any], genesisConfigFn func(*genesis.DeployConfig), jovianAtGenesis bool, minBaseFee uint64) {
+		t := actionsHelpers.NewDefaultTesting(gt)
+
+		env := helpers.NewL2FaultProofEnv(t, testCfg, helpers.NewTestParams(), helpers.NewBatcherCfg(), genesisConfigFn)
+
+		t.Logf("L2 Genesis Time: %d, JovianTime: %d ", env.Sequencer.RollupCfg.Genesis.L2Time, *env.Sequencer.RollupCfg.JovianTime)
+
+		if jovianAtGenesis {
+			// Verify Jovian is active at genesis
+			require.True(t, env.Sequencer.RollupCfg.IsJovian(env.Sequencer.RollupCfg.Genesis.L2Time), "Jovian should be active at genesis")
+		} else {
+			// If Jovian is not activated at genesis, build some blocks up to the activation block
+			// and verify that the extra data is Holocene
+			for env.Engine.L2Chain().CurrentBlock().Time < *env.Sequencer.RollupCfg.JovianTime {
+				b := env.Engine.L2Chain().GetBlockByHash(env.Sequencer.L2Unsafe().Hash)
+				expectedHoloceneExtraData := eip1559.EncodeHoloceneExtraData(250, 6)
+				require.Equal(t, expectedHoloceneExtraData, b.Extra(), "extra data should match Holocene format")
+				env.Sequencer.ActL2EmptyBlock(t)
+			}
+		}
+
+		// Build the activation block
+		env.Sequencer.ActL2EmptyBlock(t)
+		activationBlock := env.Engine.L2Chain().GetBlockByHash(env.Sequencer.L2Unsafe().Hash)
+		require.Equal(t, eip1559.EncodeJovianExtraData(250, 6, 0), activationBlock.Extra(), "activation block should have Jovian extraData")
+		require.Greater(t, activationBlock.BaseFee().Uint64(), uint64(0), "activation block should have a base fee above the minimum base fee since it was just enabled")
+
+		// Set the minimum base fee
+		setMinBaseFeeViaSystemConfig(t, env, minBaseFee)
+
+		// Allow L1->L2 derivation to propagate the SystemConfig change
+		env.Sequencer.ActL1HeadSignal(t)
+		env.Sequencer.ActL2PipelineFull(t)
+
+		// Build L2 blocks up to the L1 origin that includes the SystemConfig change
+		env.Sequencer.ActBuildToL1Head(t)
+
+		// Build activation+1 block
+		env.Sequencer.ActL2EmptyBlock(t)
+		nextBlock := env.Engine.L2Chain().GetBlockByHash(env.Sequencer.L2Unsafe().Hash)
+		actualMinBaseFee := binary.BigEndian.Uint64(nextBlock.Extra()[9:17])
+
+		expectedJovianExtraDataWithMinFee := eip1559.EncodeJovianExtraData(250, 6, minBaseFee)
+		require.Equal(t, minBaseFee, actualMinBaseFee, "minimum base fee should be equal to the set minimum base fee")
+		require.Equal(t, expectedJovianExtraDataWithMinFee, nextBlock.Extra(), "block should have updated Jovian extraData with min base fee")
+
+		// Simulate some user transactions after to ensure base fee >= minimum base fee
+		cl := env.Engine.EthClient()
+		signer := types.LatestSigner(env.Sd.L2Cfg.Config)
+		zeroAddr := common.Address{}
+		for i := 0; i < 10; i++ {
+			nonce, err := cl.PendingNonceAt(t.Ctx(), env.Dp.Addresses.Alice)
+			require.NoError(t, err)
+
+			tx := types.MustSignNewTx(env.Dp.Secrets.Alice, signer, &types.DynamicFeeTx{
```

Comment:
IIUC here we build 10 blocks with 1 tx each. Most likely this will result in the base fee dropping (it's not enough traffic to cause it to rise). You can likely verify this manually with some logging statements. 

How about we just don't do the user transactions? If the base fee is dropping, but we can show that it does not drop below the `minBaseFee`, then I think we have shown the feature is working.

Reply:
as discussed, this test will try to follow the specs as closely as possible. that is, we just want to ensure the base fee is clamped to the MBF (aka it will go no lower than the MBF)